internal/pkg/scalable/deployments.go

```diff
@@ -27,23 +33,75 @@ type deployment struct {
 	*appsv1.Deployment
 }
 
-// SetReplicas sets the amount of replicas on the resource. Changes won't be made on kubernetes until update() is called
-func (d deployment) SetReplicas(replicas int) {
+// setReplicas sets the amount of replicas on the resource. Changes won't be made on kubernetes until update() is called
+func (d deployment) setReplicas(replicas int) error {
+	if replicas > math.MaxInt32 || replicas < math.MinInt32 {
```

Comment:
I'd probably check for if the replicas is < 0 thought, as that is the lowest valid value. If you do so also change the error message below. Also new errors (errors which don't wrap another error) should be defined as a var at the top of the file (we will change those to custom errors later).